### PR TITLE
👷‍♀️FIX: expanded link ids

### PR DESCRIPTION
**Pull request by kenjinp**
Fixes a problem where some links would be missing because of SPAQL shenanigans. 

Link methods now require as a param an expanded `@id` instead of the unresolved `@id` that you would normally get as a getByID response from the server. 

the instance methods to get links will fetch the expanded resource if it wasn't expanded already in order to get this expanded `@id`.

@julienmachon you will be happy to know that this resolves the missing links betwixt your inventory of Roux sauces.

### CODE REVIEW

File: src/Resource/__tests__/links.spec.ts [hunk 1]
```diff
@@ -188,7 +192,39 @@ describe('Incoming / Outgoing Links behavior', () => {
...
+      fetchMock.resetMocks();
+    });
 
+    it('should work having already the expanded resource', async () => {
+      const resource = new Resource<{
+        subject: string;
+      }>('testOrg', 'testProject', mockGetByIDResourceResponse);
+      resource.expanded = {
+        '@id': 'https://myExpandedResourceURl.com/someID',
+      } as any;
```
Comment (julienmachon) [inline src/Resource/__tests__/links.spec.ts]: I guess we could type` ExpandedResource` as we now it have `@id`, `@type` and `[keys: string]: any`

File: src/Resource/index.ts [hunk 2]
```diff
@@ -160,24 +160,42 @@ export default class Resource<T = {}> {
...
 
+  async getExpanded() {
+    this.expanded = await getSelfResourceRawAs(`${this.self}?format=expanded`);
+  }
+
   async getIncomingLinks(
     paginationSettings: PaginationSettings,
   ): Promise<PaginatedList<ResourceLink>> {
+    let expandedID;
+    if (this.expanded && (this.expanded as any)['@id']) {
```
Comment (julienmachon) [inline src/Resource/index.ts]: We should definitely type `expanded`, I don't like this `as any`, it makes me sad

Comment (kenjinp) [reply src/Resource/index.ts]: I don't want to make you sad 😢

File: src/Resource/__tests__/links.spec.ts (hunk 1 above)
Comment (kenjinp) [reply src/Resource/__tests__/links.spec.ts]: you got it buddy

File: src/Resource/types.ts [hunk 3]
```diff
@@ -87,3 +87,8 @@ export interface ResourceLink {
   isExternal: boolean;
   predicate: string; // url vocab link such as https://myNexusInstance.com/vocabs/myOrg/friendProject/friend
 }
+
+export interface ExpandedResource {
+  '@id': string;
+  [uriKey: string]: any;
```
Comment (julienmachon) [inline src/Resource/types.ts]: what about `@type?: string[]` ?

File: src/Resource/__tests__/links.spec.ts (hunk 1 above)
Comment (julienmachon) [inline src/Resource/__tests__/links.spec.ts]: Shouldn't this be `resource.expanded: ExpandedResource` ?

File: src/Resource/types.ts (hunk 3 above)
Comment (kenjinp) [reply src/Resource/types.ts]: its not required

File: src/Resource/__tests__/links.spec.ts (hunk 1 above)
Comment (kenjinp) [reply src/Resource/__tests__/links.spec.ts]: dang good catch

File: src/Resource/types.ts (hunk 3 above)
Comment (kenjinp) [reply src/Resource/types.ts]: ah yeah it's optional, good point!

**Pull request merged**